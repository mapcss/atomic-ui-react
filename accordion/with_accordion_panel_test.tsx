import {
  anyBoolean,
  anyFunction,
  anyNumber,
  anyString,
  assertSnapshot,
  describe,
  expect,
  fn,
  it,
  setupJSDOM,
} from "../dev_deps.ts";
import WithAccordionPanel from "./with_accordion_panel.ts";
import { render } from "@testing-library/react";

const describeTests = describe({
  name: "WithAccordionPanel",
  async beforeEach() {
    await setupJSDOM();
  },
});

it(describeTests, "should render as", (t) => {
  const mockFn = fn();
  const { container } = render(
    <>
      <WithAccordionPanel
        contexts={{
          headerId: "header",
          index: 0,
          openIndex: 0,
          setOpenIndex: () => {},
          id: "panel",
        }}
      >
        {(attrs, contexts) => {
          mockFn(attrs);
          mockFn(contexts);
          return <div {...attrs}>test1</div>;
        }}
      </WithAccordionPanel>
    </>,
  );

  assertSnapshot(t, container.innerHTML);

  expect(mockFn).toHaveBeenCalledWith({
    "hidden": false,
    "aria-labelledby": "header",
    id: "panel",
  });

  expect(mockFn).toHaveBeenCalledWith({
    isOpen: anyBoolean(),
    index: anyNumber(),
    openIndex: anyNumber(),
    setOpenIndex: anyFunction(),
    id: anyString(),
    headerId: anyString(),
  });
});
