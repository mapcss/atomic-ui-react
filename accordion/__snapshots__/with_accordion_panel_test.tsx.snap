export const snapshot = {};

snapshot[`WithAccordionPanel > should render as 1`] = `'<div aria-labelledby="header" id="panel">test1</div>'`;
